### PR TITLE
Iox #590 Rename utils to hoofs

### LINKED ISSUE #794: Enforce unix line endings
## Brief feature description

All files in our code base should have the same line endings.

## Detailed information

In order to prevent accidentally use windows line endings which might result in a large diff at a later time, all file shall have the unix line ending. This can be done by a `.gitattributes` file

### PR DESCRIPTION
**Pull request by mossmaurice**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

## Notes for Reviewer
* Rename iceoryx_utils to iceoryx_hoofs 

My apologies for the large pull request :chipmunk: I tried smaller steps, but eventually decided to do it in one go.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes partly #590
- Closes #794

### CODE REVIEW

File: iceoryx_examples/iceperf/iceperf_laurel.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
```
Comment (elfenpiff) [inline iceoryx_examples/iceperf/iceperf_laurel.cpp:1]: Why is this a completely new file?

This file should be either `iceperf_follower` or `iceperf_leader`. I think you forgot to merge the latest master?! Or something like that...

Comment (elBoberido) [reply iceoryx_examples/iceperf/iceperf_laurel.cpp:1]: this file is long gone. It seems it's a bad merge

Comment (mossmaurice) [reply iceoryx_examples/iceperf/iceperf_laurel.cpp:1]: Upsi, I'll check

File: README.md [hunk 2]
```diff
@@ -89,7 +89,7 @@ Please see the dedicated [README.md](tools/docker/README.md) for information on
 | iceoryx_dds                             | 4    | 4                     |                                     |
 | iceoryx_meta                            | 5    | 5                     |                                     |
 | iceoryx_posh                            | 1+, 2| 2                     | Will be split into separate targets |
-| iceoryx_utils                           | 1+   | 2                     |                                     |
+| iceoryx_hoofs                           | 1+   | 2                     |                                     |
 | iceoryx_introspection                   | 5    | 5                     |                                     |
```
Comment (elBoberido) [inline README.md:93]: The alphabetical ordering is now even more in a muddle. Shall we order them semantically from building blocks to extensions?

File: iceoryx_hoofs/README.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 
-# Eclipse iceoryx utils overview
+# Eclipse iceoryx hoofs overview
 
-The iceoryx utils are our basic building blocks - the foundation of
+The iceoryx hoofs (**H**andy **O**bjects **O**ptimised **F**or **S**afety) are our basic building blocks - the foundation of
```
Comment (elBoberido) [inline iceoryx_hoofs/README.md:4]: Is this acronym correct? I read a different one in one of the previous files ... don't know which one, there are just too many

File: iceoryx_hoofs/include/iceoryx_hoofs/concurrent/lockfree_queue.hpp [hunk 4]
```diff
@@ -1,133 +1,134 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_CONCURRENT_LOCKFREE_QUEUE_HPP
+#define IOX_HOOFS_CONCURRENT_LOCKFREE_QUEUE_HPP
+
```
Comment (elBoberido) [inline iceoryx_hoofs/include/iceoryx_hoofs/concurrent/lockfree_queue.hpp:20]: This looks suspicious? Why is the whole file marked as changed? Line endings?

File: iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/acl.hpp [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_LINUX_PLATFORM_ACL_HPP
+#define IOX_HOOFS_LINUX_PLATFORM_ACL_HPP
```
Comment (elBoberido) [inline iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/acl.hpp:18]: why is this whole file marked as new

File: iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/dlfcn.hpp [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_LINUX_PLATFORM_DLFCN_HPP
+#define IOX_HOOFS_LINUX_PLATFORM_DLFCN_HPP
```
Comment (elBoberido) [inline iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/dlfcn.hpp:18]: same here

File: iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/getopt.hpp [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_LINUX_PLATFORM_GETOPT_HPP
+#define IOX_HOOFS_LINUX_PLATFORM_GETOPT_HPP
```
Comment (elBoberido) [inline iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/getopt.hpp:18]: same here and with all the other files from platform?

**Review by elBoberido — COMMENTED** [review]
First 284 files. The remaining ones will follow

File: iceoryx_hoofs/include/iceoryx_hoofs/concurrent/lockfree_queue.hpp (hunk 4 above)
Comment (mossmaurice) [reply iceoryx_hoofs/include/iceoryx_hoofs/concurrent/lockfree_queue.hpp:20]: Exactly line ending was Windows (CRLF) and is now Linux (LF)

File: iceoryx_hoofs/README.md (hunk 3 above)
Comment (mossmaurice) [reply iceoryx_hoofs/README.md:4]: Jep, see [here](https://github.com/eclipse-iceoryx/iceoryx/issues/482#issuecomment-791323674)

File: iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/acl.hpp (hunk 5 above)
Comment (mossmaurice) [reply iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/acl.hpp:18]: TBH, I have no clue :( I've moved all the files with the same `git mv` command. I know that Git has some magic detection mechanisms to determine how many percent of a file has changed. I don't understand why this worked with the other platform files, though.

Double checked, no duplicates around:
```
./iceoryx_hoofs/platform/qnx/include/iceoryx_hoofs/platform/acl.hpp
./iceoryx_hoofs/platform/mac/include/iceoryx_hoofs/platform/acl.hpp
./iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/acl.hpp
./iceoryx_hoofs/platform/win/include/iceoryx_hoofs/platform/acl.hpp
```

File: iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/dlfcn.hpp (hunk 6 above)
Comment (mossmaurice) [reply iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/dlfcn.hpp:18]: See [here](https://github.com/eclipse-iceoryx/iceoryx/pull/790/files#r630935855)

File: iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/getopt.hpp (hunk 7 above)
Comment (mossmaurice) [reply iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/getopt.hpp:18]: See [here](https://github.com/eclipse-iceoryx/iceoryx/pull/790/files#r630935855)

**Review by elBoberido — COMMENTED** [review]
400 files done and I just noticed you didn't run clang-format

File: iceoryx_hoofs/README.md (hunk 3 above)
Comment (elBoberido) [reply iceoryx_hoofs/README.md:4]: can we find an acronym where `H` stands for hypnotoad?

Comment (elfenpiff) [reply iceoryx_hoofs/README.md:4]: I think this would be blasphemy, using Hypnotoad as a helper or a name for building blocks ;). We all praise Hypnotoad and work for it and not the other way around

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/list.inl [hunk 8]
```diff
@@ -529,7 +529,7 @@ inline bool list<T, Capacity>::IteratorBase<IsConstIterator>::operator!=(
 template <typename T, uint64_t Capacity>
 template <bool IsConstIterator>
 inline typename list<T, Capacity>::template IteratorBase<IsConstIterator>::reference
-    list<T, Capacity>::IteratorBase<IsConstIterator>::operator*() const noexcept
+list<T, Capacity>::IteratorBase<IsConstIterator>::operator*() const noexcept
```
Comment (elfenpiff) [inline iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/list.inl:532]: Oh, here is a very important `inline` missing. Could you please add this otherwise someone will get sooner or later a double definition error during linking.

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/list.inl [hunk 9]
```diff
@@ -538,7 +538,7 @@ inline typename list<T, Capacity>::template IteratorBase<IsConstIterator>::refer
 template <typename T, uint64_t Capacity>
 template <bool IsConstIterator>
 inline typename list<T, Capacity>::template IteratorBase<IsConstIterator>::pointer
-    list<T, Capacity>::IteratorBase<IsConstIterator>::operator->() const noexcept
+list<T, Capacity>::IteratorBase<IsConstIterator>::operator->() const noexcept
```
Comment (elfenpiff) [inline iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/list.inl:541]: Oh, here is a very important `inline` missing. Could you please add this otherwise someone will get sooner or later a double definition error during linking.

File: iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/acl.hpp (hunk 5 above)
Comment (elfenpiff) [reply iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/acl.hpp:18]: Did you call `git mv iceoryx_utils iceoryx_hoofs` or did you do it for every single file?

**Review by elBoberido — COMMENTED** [review]
600 filed done ... time for lunch

File: iceoryx_integrationtest/Readme.md [hunk 10]
```diff
@@ -51,7 +51,7 @@ iceoryx_workspace
         ├── iceoryx_meta
         ├── iceoryx_posh
         ├── iceoryx_integrationtest
-        ├── iceoryx_utils
+        ├── iceoryx_hoofs
```
Comment (elBoberido) [inline iceoryx_integrationtest/Readme.md:54]: only if you have to do a commit, could you reorder this alphabetically

**Review by elBoberido — COMMENTED** [review]
done

File: iceoryx_hoofs/platform/qnx/include/iceoryx_hoofs/platform/un.hpp [hunk 11]
```diff
@@ -1,4 +1,5 @@
 // Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
```
Comment (elfenpiff) [inline iceoryx_hoofs/platform/qnx/include/iceoryx_hoofs/platform/un.hpp:2]: This is weird? Why did you move `wait.hpp` to `un.hpp`?
Where is the original `wait.hpp` file?! Or did here something else go wrong ... I am a little bit lost

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/list.inl (hunk 9 above)
Comment (mossmaurice) [reply iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/list.inl:541]: Please have a look two lines above ;)

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/list.inl (hunk 8 above)
Comment (mossmaurice) [reply iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/list.inl:532]: Please have a look two lines above ;)

**Review by elfenpiff — COMMENTED** [review]
Could you please replace `#pragma once`  with include guards in iceoryx_hoofs/platform/win/include/iceoryx_hoofs/platform/platform_correction.hpp

File: iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/acl.hpp (hunk 5 above)
Comment (elBoberido) [reply iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/acl.hpp:18]: I think this is just the github heuristic which is doesn't recognize that it is an old file. I looked at the history of the file and it is preserved

File: iceoryx_hoofs/source/log/logging_internal.cpp [hunk 12]
```diff
@@ -1,4 +1,5 @@
 // Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
```
[the remark this answers is not in the document]
Comment (mossmaurice) [reply iceoryx_hoofs/source/log/logging_internal.cpp:2]: Fixed

File: iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/acl.hpp (hunk 5 above)
Comment (mossmaurice) [reply iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/acl.hpp:18]: The former. Maybe I did something wrong when adding the files.

File: iceoryx_hoofs/platform/qnx/include/iceoryx_hoofs/platform/un.hpp (hunk 11 above)
Comment (mossmaurice) [reply iceoryx_hoofs/platform/qnx/include/iceoryx_hoofs/platform/un.hpp:2]: That's odd, indeed. Shoulnd't be an issue since `wait.hpp` and `un.hpp` are definitely there. I have the feeling I need to practice [this dance](https://www.youtube.com/watch?v=hdcTmpvDO0I) more thoroughly.

File: tools/HelixQAC/README.md [hunk 13]
```diff
@@ -20,7 +20,7 @@ with the compiler command being a command used to build an iceoryx file. If you
 
 an example command if you use GCC (command taked from the build log):
 
-    CCT_Generator.sh -c /usr/bin/c++  -Diceoryx_utils_EXPORTS -I/home/perforce/iceoryx/iceoryx_utils/include -I/home/perforce/iceoryx/iceoryx_utils/platform/linux/include  -O3 -DNDEBUG -fPIC   -W -Wall -Wextra -Wuninitialized -Wpedantic -Wstrict-aliasing -Wcast-align -Wno-noexcept-type -Wconversion -std=gnu++14 -o CMakeFiles/iceoryx_utils.dir/source/posix_wrapper/unix_domain_socket.cpp.o -c /home/perforce/iceoryx/iceoryx_utils/source/posix_wrapper/unix_domain_socket.cpp
+    CCT_Generator.sh -c /usr/bin/c++  -Diceoryx_hoofs_EXPORTS -I/home/perforce/iceoryx/iceoryx_hoofs/include -I/home/perforce/iceoryx/iceoryx_hoofs/platform/linux/include  -O3 -DNDEBUG -fPIC   -W -Wall -Wextra -Wuninitialized -Wpedantic -Wstrict-aliasing -Wcast-align -Wno-noexcept-type -Wconversion -std=gnu++14 -o CMakeFiles/iceoryx_hoofs.dir/source/posix_wrapper/unix_domain_socket.cpp.o -c /home/perforce/iceoryx/iceoryx_hoofs/source/posix_wrapper/unix_domain_socket.cpp
```
Comment (elfenpiff) [inline tools/HelixQAC/README.md:23]: Does this actually work, is this somewhere tested? I mean where is this `-Diceoryx_hoofs_EXPORTS` defined since I do not find the corresponding `CCT_Generator.sh` script in our repo I assume the script uses something from us where we define that compiler variable?!

Comment (mossmaurice) [reply tools/HelixQAC/README.md:23]: It was tested in #757 by @dkroenke @marthtz and @toniglandy1. IIUC these are compiler switches from CMake.

Comment (dkroenke) [reply tools/HelixQAC/README.md:23]: @elfenpiff 
`-D<libname>_EXPORTS` is automatically generated by cmake when building a shared library. 

From https://cmake.org/cmake/help/v3.0/command/set_target_properties.html:
> DEFINE_SYMBOL sets the name of the preprocessor symbol defined when compiling sources in a shared library. If not set here then it is set to target_EXPORTS by default (with some substitutions if the target is not a valid C identifier). This is useful for headers to know whether they are being included from inside their library or outside to properly setup dllexport/dllimport decorations.

**Review by elBoberido — APPROVED** [review]

**Review by elfenpiff — APPROVED** [review]
